app/components/GameBoard.tsx

```
@@ -1,66 +1,304 @@
+      const mouse = new THREE.Vector2(
+        ((event.clientX - rect.left) / rect.width) * 2 - 1,
+        -((event.clientY - rect.top) / rect.height) * 2 + 1
+      );
+
+      const raycaster = new THREE.Raycaster();
+      raycaster.setFromCamera(mouse, cameraRef.current);
+      const intersects = raycaster.intersectObjects(cellMeshesRef.current);
+
+      if (intersects.length > 0) {
+        const intersectedCell = intersects[0].object;
+        if (intersectedCell.userData && intersectedCell.userData.isCell) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)